### ISSUE TITLE
Inconsistent time behavior between popup and textbox 

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-pickers#2043]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui-pickers/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

The time depend on the usage of the popup or textbox

## Expected Behavior 🤔

Consistency

## Steps to Reproduce 🕹

Steps:

1. Open https://codesandbox.io/s/old-waterfall-m6mxw?file=/src/index.js
2. Follow the steps of https://github.com/mui-org/material-ui-pickers/issues/1526#issuecomment-663821463

| Tech        | Version |
| ----------- | ------- |
| @material-ui/pickers | v4.0.0-alpha.10  |

## Context

Raised by @dakshsaraf in https://github.com/mui-org/material-ui-pickers/issues/1526#issuecomment-663821463

**Comment by BrandonWilliamsCS** [comment]
@oliviertassinari, I believe this issue and the "parent" issue #1526 are actually pretty closely connected - if not simply two perspectives on the same issue. Solving one will probably solve the other. My comment [here](https://github.com/mui-org/material-ui-pickers/issues/1526#issuecomment-754935608) explains what I mean, but the most relevant idea is that the behavior of the pop-up version is actually _incorrect_ and not just _inconsistent_ with the keyboard version's behavior.

In short, when fixing this issue, I strongly suggest making sure the time of the returned value is midnight in both cases rather than the "current time" in both cases.
